packages/node-http-handler/src/node-http2-handler.spec.ts

```diff
@@ -28,288 +29,420 @@ describe("NodeHttp2Handler", () => {
-      ).toBeDefined();
+    it("has metadata", () => {
+      expect(nodeH2Handler.metadata.handlerProtocol).toContain("h2");
     });
 
-    it("reuses existing session if request is made on same authority again", async () => {
-      await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
-      // @ts-ignore: access private property
-      expect(nodeH2Handler.connectionPool.size).toBe(1);
+    describe("number calls to http2.connect", () => {
+      it("is zero on initialization", () => {
+        const connectSpy = jest.spyOn(http2, "connect");
+        expect(connectSpy).not.toHaveBeenCalled();
+      });
+
+      it("is one when request is made", async () => {
+        const connectSpy = jest.spyOn(http2, "connect");
+
+        // Make single request.
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const authority = `${protocol}//${hostname}:${port}`;
+        expect(connectSpy).toHaveBeenCalledTimes(1);
+        expect(connectSpy).toHaveBeenCalledWith(authority);
+      });
+
+      it("is one if multiple requests are made on same URL", async () => {
+        const connectSpy = jest.spyOn(http2, "connect");
 
-      // @ts-ignore: access private property
-      const session: ClientHttp2Session = nodeH2Handler.connectionPool.get(`${protocol}//${hostname}:${port}`);
-      const requestSpy = jest.spyOn(session, "request");
+        // Make two requests.
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
 
-      await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
-      // @ts-ignore: access private property
-      expect(nodeH2Handler.connectionPool.size).toBe(1);
-      expect(requestSpy.mock.calls.length).toBe(1);
+        const authority = `${protocol}//${hostname}:${port}`;
+        expect(connectSpy).toHaveBeenCalledTimes(1);
+        expect(connectSpy).toHaveBeenCalledWith(authority);
+      });
+
+      it("is many if requests are made on different URLs", async () => {
+        const connectSpy = jest.spyOn(http2, "connect");
+
+        // Make first request on default URL.
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const port2 = port + 1;
+        const mockH2Server2 = createMockHttp2Server().listen(port2);
+        mockH2Server2.on("request", createResponseFunction(mockResponse));
+
+        // Make second request on URL with port2.
+        await nodeH2Handler.handle(new HttpRequest({ ...getMockReqOptions(), port: port2 }), {});
+
+        const authorityPrefix = `${protocol}//${hostname}`;
+        expect(connectSpy).toHaveBeenCalledTimes(2);
+        expect(connectSpy).toHaveBeenNthCalledWith(1, `${authorityPrefix}:${port}`);
+        expect(connectSpy).toHaveBeenNthCalledWith(2, `${authorityPrefix}:${port2}`);
+        mockH2Server2.close();
+      });
```

Comment:
Another unit test for `"is many for same URL but concurrent stream is disabled"`

Reply:
The concurrent stream disabled tests are in their own section on line 378.
The section starts with `describe("disableConcurrentStreams", () => {`

The specific test mentioned `is many for same URL but concurrent stream is disabled` is called `is many if multiple requests are made on same URL`, and is on line 402.